bin/check.ml

```diff
@@ -20,15 +20,18 @@ let check (`Package_names pkg_names) (`Package_version version) (`Dist_tag tag)
      if on_working_tree then (OS.Dir.current (), fun _ -> ())
      else
        let dir =
-         Pkg.tag_from_repo ?tag ?version () >>= fun inferred_tag ->
+         (* TODO this is a weird cludge because `Pkg.tag` wants a `Pkg.t` *)
```

Comment:
I think the check command should likely create its own `Pkg.t` based on the CLI arguments, just as all other commands do.

The problem is that the `Pkg.t` type isn't super well defined and it in facts represent an abstraction of the repository rather than a single package. This is something we'll work on improving shortly for 2.0 but for now I think it's okay to just properly initialize one and pass it around same as in the other commands.

Reply:
Thanks. The whole logic in `check` is rather complex, with `Pkg.v` created in various places in seemingly hackish ways so I guess this makes sense until the whole logic is revisited.